src/main/js/common/api.js

```diff
@@ -4,42 +4,35 @@
 
 import { getJSON, postJSON, post } from "sonar-request";
 
-// Function used to get current SonarQube Server version
+//Function used to get current SonarQube Server version
 function getSonarVersion() {
   return getJSON("/api/system/status").then(response => {
     return parseFloat(response.version);
   });
 }
 
-// Function used to get the list of existing projects (until 7.9.x)
-function getProjectsList79() {
-  return getJSON("/api/projects/index").then(response => {
-    return response.map(item => {
-      // Attribute names changed in SonarQube 8.0
-      // so we need to match the new ones for the frontend
-      item.key = item.k;
-      item.name = item.nm;
-      return item;
-    });
-  });
-}
-
-// Function used to get the list of existing projects (since 8.0)
-function getProjectsList8() {
-  return getJSON("/api/projects/search").then(response => {
-    return response.components;
-  });
-}
-
-// Function used to get the list of existing project regarding SonarQube version
-export function getProjectsList() {
-  return getSonarVersion().then(version => {
-    // Depending on SonarQube version, the API is not the same to retrieve projects list
-    if (version >= 8.0) {
-      return getProjectsList8();
-    } else {
-      return getProjectsList79();
+//Functions used to get all user projects
+export function getProjectsList(){
+  let elementByPage = 500;
+  let allPromises = [];
+  let finalProjectsList = [];
+  return getJSON("/api/components/search", {"qualifiers": "TRK", "ps":elementByPage}).then(response => {
+    let nbProjects = response.paging.total;
+    let nbPages = Math.ceil(nbProjects/elementByPage);
+    for(let i = 1; i <= nbPages; i++){
+      allPromises.push(getJSON("/api/components/search", {"qualifiers": "TRK", "ps":elementByPage, "p":i}).then(response => {
+        return response.components;
+      }));
     }
+    return Promise.all(allPromises).then((results) => {

```

Comment:
In order to avoid having too much nesting "then" methods, I think you should move this one and put it right after the previous "then".
So that you would have "getJson" -> then (returning Promise.all) -> then (to return concatenated results).
What do you think ?

Reply:
I don't really understand your idea.
I prefer not to touch this promise because I do not control its scope at 100% for the moment.